### DISCUSSION TITLE
Transcription via OpenAi's whisper: AssertionError: incorrect audio shape

### DISCUSSION THREAD

**Discussion by murasz** [openai/whisper#1099] [Q&A] [answered]
I'm trying to use OpenAI's open source Whisper library to transcribe audio files.

Here is my script's source code:

```
import whisper

model = whisper.load_model("large-v2")

# load the entire audio file
audio = whisper.load_audio("/content/file.mp3")
#When i write that code snippet here ==> audio = whisper.pad_or_trim(audio) the first 30 secs are converted and without any problem they are converted.

# make log-Mel spectrogram and move to the same device as the model
mel = whisper.log_mel_spectrogram(audio).to(model.device)

# detect the spoken language
_, probs = model.detect_language(mel)
print(f"Detected language: {max(probs, key=probs.get)}")

# decode the audio
options = whisper.DecodingOptions(fp16=False)
result = whisper.decode(model, mel, options)

# print the recognized text if available
try:
    if hasattr(result, "text"):
        print(result.text)
except Exception as e:
    print(f"Error while printing transcription: {e}")

# write the recognized text to a file
try:
    with open("output_of_file.txt", "w") as f:
        f.write(result.text)
        print("Transcription saved to file.")
except Exception as e:
    print(f"Error while saving transcription: {e}")
```
In here:

```
# load the entire audio file
audio = whisper.load_audio("/content/file.mp3")
```
when I write below: " audio = whisper.pad_or_trim(audio) ", the first 30 secs of the sound file is transcribed without any problem and language detection works as well,

but when I delete it and want the whole file to be transcribed, I get the following error:

`AssertionError: incorrect audio shape`

What should I do? Should I change the structure of the sound file? If yes, which library should I use and what type of script should I write? I'll be glad for help. Thanks!

**Comment by jongwook** [comment, marked as answer]
Hi, please try `transcribe()` function which will chunk the audio into sliding 30-second windows and return the full transcript.

**Reply by murasz** [reply]
Where should I place  ```transcribe()``` function exactly in the code snippet? Thanks.

**Reply by jongwook** [reply]
Please try replacing the line

```
audio = whisper.load_audio("/content/file.mp3")
```

and below, with:

```
result = model.transcribe("/content/file.mp3")
print(result)
```

**Reply by murasz** [reply]
> Please try replacing the line
> 
> ```
> audio = whisper.load_audio("/content/file.mp3")
> ```
> 
> and below, with:
> 
> ```
> result = model.transcribe("/content/file.mp3")
> print(result)
> ```

Hi Jong, 

Thanks for the information. I finally managed to run the script without any errors after your suggestion and basically simplifying the code snippet. The only funny thing is that although what I am trying to transcribe is in Turkish language, due to the accent of the speaking person in the file, it detects Azeri instead of Turkish lol. Is there a way to force Whisper to transcribe a file in a specific language without it automatically trying to detect the language and implement it? 

Btw, for those, who have the same type of errors, you can use the code below:

```
import whisper
import os

# input audio file name
audio_file = "/content/file.mp3"

# load the model and transcribe the audio
model = whisper.load_model("large-v2")
result = model.transcribe(audio_file)

# extract the text and language information
text = result["text"]
language = result["language"]

# create the output text file name based on the input mp3 file name
text_file = os.path.splitext(audio_file)[0] + ".txt"

# write the text and language information to the output text file
with open(text_file, "w") as f:
    f.write(f"Text:\n\n{text}\n\nLanguage: {language}")

# print the text and language information to the console
print("Text:\n\n", text)
print("Language: ", language)
```

**Reply by jongwook** [reply]
Language detection is not very accurate unfortunately, and may result in something like what you're experiencing. If you know the audio is in Turkish, you can specify in the `transcribe()` call to skip the language detection part and use that language, like:

```python
result = model.transcribe("/content/file.mp3", language="Turkish")
```

**Reply by murasz** [reply]
Thank you!
